lib/msal-core/src/utils/ResponseUtils.ts

```diff
@@ -31,4 +34,28 @@ export class ResponseUtils {
             tenantId: idTokenObj.tenantId,
         };
     }
+
+    static buildAuthResponse(idToken: IdToken, authResponse: AuthResponse, serverAuthenticationRequest: ServerRequestParameters, account: Account, scopes: Array<string>, accountState: string): AuthResponse {
+        switch(serverAuthenticationRequest.responseType) {
+            case ResponseTypes.id_token:
+                authResponse = {
+                    ...authResponse,
+                    tokenType: ServerHashParamKeys.ID_TOKEN,
+                    account: account,
+                    scopes: scopes,
+                    accountState: accountState
+                };
+                
+                authResponse = ResponseUtils.setResponseIdToken(authResponse, idToken);
+                return (authResponse.idToken) ? authResponse : null;
+            case ResponseTypes.id_token_token:
+                authResponse = ResponseUtils.setResponseIdToken(authResponse, idToken);
+                return (authResponse && authResponse.idToken) ? authResponse : null;
```

Comment:
should we also be checking that `authResponse.accessToken` is truthy here?

Reply:
Adding the check, thanks for catching it. I'm fairly certain authResponse would be `null` if there was no access token, then `setResponseIdToken` would return null immediately if that was the case, making this line correctly return `null` in the end. However, I think it's a good idea to code defensively against future changes in `getCachedToken` that may not be caught by the current logic.